toc.md

```diff
@@ -428,77 +429,234 @@ An additional backlog of 936 hours of image processing jobs everyday, along with
+
+We benchmarked the memory consumption of having one process and two processes by using a Rails application. Having one process takes up 125MB of memory, but having two processes don't take twice as much memory. This is the copy-on-write mechanism at work.
 
 ![memory_usage_processes](/images/memory_usage_processes.png)
 
-Here, we benchmarked both having one process and two processes. Having one process takes up 125MB of memory, but having two processes don't take twice as much memory. This is the copy-on-write mechanism at work.
+Let's assume the Rails application in our scenario is a pretty big Rails app and takes about 400MB. By forking 4 processes Workerholic, will at most take 1.6GB of memory out of 4GB, meaning, forking 4 processes is a viable option.
 
-As we mentioned previously, using our four CPU cores, we can fork to a total of four processes and reduce our image processing enqueuing:processing ratio down to 1:10. But that is still not good enough. After an extended period of time, we will eventually end up with a backlog and a huge memory footprint. Where do we go from here?
+As we mentioned previously, using our four CPU cores, we can fork to a total of four processes and reduce our image processing enqueuing:processing ratio down to 1:10. But that is still not good enough. After an extended period of time, we will eventually end up with a backlog of jobs, effectively implicating a huge latency and memory footprint. What else can we do to even out or enqueuing:processing ratio?
 
 ### Scalability
 
-#### Scaling in the context of our scenario
 ![scalibility_image](/images/scalibility_image.png){:width="380" height="350"}
 
-Currently, we are using all of our resources on hand to evenly processing throughput, but with an enqueuing:processing ratio of 1:10. It is still not good enough. At this point, we have to scale our system. We have two options:
+#### Scaling in the context of our scenario
+
+We are currently maximizing the use of available resources by using concurrency and parallelism. In order to bring our enqueuing:processing down to a 1:1 ratio we need to scale either vertically or horizontally.
 
 ##### Scaling vertically
-We can get more cores on a single machine. If we get 20 cores, we still have a 1:2 enqueuing:processing ratio, so we will need 40 cores. This does not sound very plausible. Have you ever heard of a single machine having this many cores?
+
+Scaling vertically means increasing the amount of available resources by getting a better machine. We currently have 4 CPU cores and 4GB or RAM. By looking at [Digital Ocean](https://www.digitalocean.com/pricing/#droplet), their best machine is 20 CPU cores and 64GB or RAM. If we were to use one these machine instead of the current configuration we have we would be able to bring our enqueuing:processing ratio down to 1:2. The memory footprint for 20 processes will be of 8GB *(20 * 400MB)* at worst, which is acceptable since this machine would have 64GB of RAM.
```

Comment:
Scaling vertically means increasing the amount of available resources by getting a better machine. We currently have 4 CPU cores and 4GB of RAM. By looking at [Digital Ocean](https://www.digitalocean.com/pricing/#droplet), their best machine has  20 CPU cores and 64GB of RAM. If we were to use one of these machine instead of what we have right now, we would be able to bring our enqueuing:processing ratio down to 1:2. The memory footprint for 20 processes will be of 8GB *(20 * 400MB)* at worst, which is acceptable since this machine would have 64GB of RAM.